src/OVAL/probes/independent/system_info.c

```diff
@@ -295,6 +305,142 @@ static int get_ifs(SEXP_t *item)
         return rc;
 }
 
+#elif defined(_WIN32)
+
+#define VERSION_LEN 32
+/* Microsoft recommends to start with a 15 kB buffer for GetAdaptersAddresses */
+#define WORKING_BUFFER_SIZE 15000
+#define MAX_TRIES 3
+
+static int get_ifs(SEXP_t *item)
+{
+	/*
+	 * Use the MAKEWORD(lowbyte, highbyte) macro declared in Windef.h
+	 * to define the Windows Socket specification version we will use.
+	 * The current version of the Windows Sockets specification is version 2.2.
+	 * Version 2.2 is supported on all systems since Windows 95 OSR2.
+	 */
+	WORD wVersionRequested = MAKEWORD(2, 2);
+	LPTSTR lpMsgBuf = NULL;
+
+	WSADATA wsaData;
+	int err = WSAStartup(wVersionRequested, &wsaData);
+	if (err != 0) {
+		FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER |
+			FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+			NULL, err, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+			lpMsgBuf, 0, NULL);
+		dE("Can't initialize Winsock DLL. WSAStartup failed. %s", lpMsgBuf);
+		return 1;
+	}
+
+	PIP_ADAPTER_ADDRESSES pAddresses = NULL;
+	PIP_ADAPTER_ADDRESSES pCurrAddresses = NULL;
+	DWORD dwRetVal = 0;
+	ULONG outBufLen = WORKING_BUFFER_SIZE;
+	ULONG family = AF_UNSPEC; // both IPv4 and IPv6
+	ULONG flags = GAA_FLAG_INCLUDE_PREFIX;
+	ULONG iterations = 0;
+
+	/* Try to get information about network adapters and their adresses */
+	do {
+		pAddresses = malloc(WORKING_BUFFER_SIZE);
+		dwRetVal = GetAdaptersAddresses(family, flags, NULL, pAddresses, &outBufLen);
+		if (dwRetVal == ERROR_BUFFER_OVERFLOW) {
+			free(pAddresses);
+			pAddresses = NULL;
+		}
+		iterations++;
+	} while (dwRetVal == ERROR_BUFFER_OVERFLOW && iterations < MAX_TRIES);
+
+	if (dwRetVal != NO_ERROR) {
+		FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER |
+			FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+			NULL, dwRetVal, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+			lpMsgBuf, 0, NULL);
+		dE("Can't get information about network adapters and their adresses. %s", lpMsgBuf);
+		free(pAddresses);
+		WSACleanup();
+		return 1;
+	}
+
+	/* Iterate over all network adapters */
+	pCurrAddresses = pAddresses;
+	while (pCurrAddresses != NULL) {
+		/* MAC address*/
+		char *mac_address_str;
+		if (pCurrAddresses->PhysicalAddressLength != 0) {
+			/* n bytes of MAC address will be printed as:
+			(2 * n) hexa numbers + (n - 1) delimiters + 1 terminating null byte = 3 * n */
+			const size_t mac_address_str_len = pCurrAddresses->PhysicalAddressLength * 3;
+			mac_address_str = malloc(mac_address_str_len);
+
+			for (int i = 0; i < (int)pCurrAddresses->PhysicalAddressLength;
```

Comment:
if length is unsigned and i is also unsigned why not just use unsigned instead of casting here?